hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java

```diff
@@ -65,10 +65,10 @@ public void preHandle(@NonNull final PreHandleContext context) {
             // If a spender who is granted approveForAll from owner and is granting
             // allowance for a serial to another spender, need signature from the approveForAll
             // spender
-            var operatorId = allowance.delegatingSpenderOrElse(ownerId);
+            var operatorId = allowance.hasDelegatingSpender() ? allowance.delegatingSpender() : ownerId;
             // If approveForAll is set to true, need signature from owner
             // since only the owner can grant approveForAll
-            if (allowance.hasApprovedForAll()) {
+            if (allowance.hasApprovedForAll() && allowance.approvedForAll().booleanValue()) {
```

Comment:
My IDE shows this as a warning on "booleanValue". Until we get our static analysis dialed in, we need to avoid these warnings. You can use approveForAllOrThrow in this case.

Reply:
Interesting. I did not understand why it shows that as warning.